ARM64: dts: ast2700-evb: Enable Uart/SPI/soc/pinctrl/irqchip/net/wdt drivers changes

ashok2ku opened the pull request:
Adding initial Commit to support Aspeed 2700 board

ashok2ku committed:
ARM64: dts: ast2700-evb: Adding linux shell boot support from AspeedTech

DTS/Device driver patches from AspeedTech-BMC/linux/tree/aspeed-master-v6.6 branch
Linux Kernel base : 6.6.36 version.

Added minimum set of features required to boot Aspeed 2700 EVB board on shell

uart/spi/soc/pinctrl/irqchip/net/wdt driver updates from AspeedTech-BMC
clock/reset and 2700 related header file
Added Aspeed tree -ast2700-evb.dts and dependent dts file

Tested: ast2700 evb machine boots to shell on evaluation board/QEMU
OpenBMC Linux base Version: 6.6.36
QEMU: 02d9c38236c

Signed-off-by: ashok2ku <Ashok2.Kumar@amd.com>

ojayanth commented:
arch/arm64/boot/dts/aspeed/Makefile  related evb dtb related changes  missing ?

It will be good to add additional information's for future reference.
Linux shell support for ast2700 eval board.
Key- features enabled : uart/spi driver updates and minimum set of features required to boot eval board on shell.
Linux Kernel base : 6.6.36 version. 
Aspped tree Reference:
Verified : details.

ashok2ku commented:
> arch/arm64/boot/dts/aspeed/Makefile related evb dtb related changes missing ?
> 
> It will be good to add additional information's for future reference. Linux shell support for ast2700 eval board. Key- features enabled : uart/spi driver updates and minimum set of features required to boot eval board on shell. Linux Kernel base : 6.6.36 version. Aspped tree Reference: Verified : details.

Added makefile and updated commit message

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts:
```diff
@@ -0,0 +1,140 @@
+	};
+};
+
+
```
Remove extra line

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts:
```diff
@@ -0,0 +1,140 @@
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+
```
Remove extra line

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts:
```diff
@@ -0,0 +1,140 @@
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+		#include "aspeed-flash-layout-128.dtsi"
```
Usually #include aligns to node. , Should align to  line 81.

ojayanth commented on arch/arm64/boot/dts/aspeed/Makefile:
```diff
@@ -1,2 +1,3 @@
 dtb-$(CONFIG_ARCH_ASPEED) += \
```
Comments Commit message :
ARM64: dts: aspeed:  Adding AMD Marley

Added base version of device tree file for AMD Marley platform based on
Aspeed 2700 SOC malta

Tested: Verified on Marley with Malta by  booting to shell.

ojayanth commented on arch/arm64/Kconfig.platforms:
```diff
@@ -33,6 +33,20 @@ config ARCH_APPLE
 	  This enables support for Apple's in-house ARM SoC family, starting
 	  with the Apple M1.
 
+config ARCH_ASPEED
```
Commit message:

ARM64: dts: ast2700-evb: Adding  linux shell  boot support

DTS/Device tree patches from AspeedTech-BMC/linux/tree/aspeed-master-v6.6 branch
Linux Kernel base : 6.6.36 version.

Added minimum set of features required to boot Aspeed 2700 EVB board on shell
- uart/spi/soc/pinctrl/irqchip/net/wdt driver updates from AspeedTech-BMC
- clock/reset and 2700 related header file
- Added Aspeed tree -ast2700-evb.dts and dependent dts file
 
Tested: ast2700 evb machine boots to shell on evaluation board/QEMU
            OpenBMC Linux base Version: 6.6.36 
            QEMU: 02d9c38236c

ashok2ku committed:
ARM64: dts: aspeed: Adding AMD Marley

Added base version of device tree file for AMD Marley platform based on
Aspeed 2700 SOC malta

Tested: Verified on Marley with Malta by booting to shell.

Signed-off-by: ashok2ku <Ashok2.Kumar@amd.com>
